### PR TITLE
Fix CI: add uv lock PyPI override to deploy_pypi workflow and regener…

### PR DESCRIPTION
**Pull request by humnabaa**
update lockfile to PyPI + edit documentation

## Description
<details><summary>Please include a summary of the changes. </summary>

  - What is this change?
  - Is this a bug fix or a feature and does it break any existing functionality?
  - How has it been tested?
</details>
 
*This pr introduces....*
## Type of change

*You can delete options that are not relevant.*

- [ ] Bug fix - *non-breaking change*
- [ ] New feature - *non-breaking change*
- [ ] Breaking change - *backwards incompatible change, changes expected behaviour*
- [ ] Non-user facing change, structural change, dev functionality, docs ...

## Checklist:

- [ ] I have performed a self-review of my own code.
- [ ] I have commented my code appropriately, focusing on explaining my design decisions (explain why, not how).
- [ ] I have made corresponding changes to the documentation (comments, docstring, etc.. )
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] New and existing unit tests pass locally with my changes.
- [ ] I have updated the change log.

<br>

#  Peer review
Any new code includes all the following:

- **Documentation**: docstrings, comments have been added/ updated.
- **Style guidelines**: New code conforms to the project's contribution guidelines.
- **Functionality**: The code works as expected, handles expected edge cases and exceptions are handled appropriately.
- **Complexity**: The code is not overly complex, logic has been split into appropriately sized functions, etc..
- **Test coverage**: Unit tests cover essential functions for a reasonable range of inputs and conditions. Added and existing tests pass on my machine.

### Review comments
Suggestions should be tailored to the code that you are reviewing. Provide context.
Be critical and clear, but not mean. Ask questions and set actions.
<details><summary>These might include:</summary>

- bugs that need fixing (does it work as expected? and does it work with other code
  that it is likely to interact with?)
- alternative methods (could it be written more efficiently or with more clarity?)
- documentation improvements (does the documentation reflect how the code actually works?)
- additional tests that should be implemented
  - Do the tests effectively assure that it
  works correctly? Are there additional edge cases/ negative tests to be considered?
- code style improvements (could the code be written more clearly?)
</details>
<br>

*Further reading: [code review best practices](https://best-practice-and-impact.github.io/qa-of-code-guidance/peer_review.html)*

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -7,6 +7,12 @@ and this project adheres to [semantic versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased]
 
+### Changed
+- Updated `deploy_pypi.yaml`, `pull_request_workflow.yaml`, and `deploy_mkdocs.yaml` to re-solve dependencies against PyPI before `uv sync`/`uv build` using `uv lock --default-index https://pypi.org/simple`, fixing CI failures caused by the internal Artifactory index not being accessible from GitHub Actions.
+- Regenerated `uv.lock` to use PyPI package URLs instead of internal Artifactory URLs.
+- Reverted unintentional dict comprehension change in `test_utils.py` back to `dict.fromkeys(...)`.
+- Applied `ruff-format` to `helpers/pyspark_log_parser/parser.py` to fix pre-commit formatting check.
```
Comment (coatet) [inline CHANGELOG.md]: Can you remove these from the changelog? No need to document these!

File: README.md [hunk 2]
```diff
@@ -22,7 +22,13 @@ The RDSA team sits within the Economic Statistics Change Directorate, and uses c
 
 `rdsa-utils` is available for installation via [PyPI](https://pypi.org/project/rdsa-utils/) and can also be found on [GitHub Releases](https://github.com/ONSdigital/rdsa-utils/releases) for direct downloads and version history.
 
-To install via `pip`, simply run:
+Install with `uv`:
```
Comment (coatet) [inline README.md]: Just a little change to wording:

```suggestion
Add to your package's dependencies with `uv`:
```

File: uv.lock
Comment (coatet) [inline uv.lock:1]: The uv.lock is for rdsa-utils developers so it should point to Artifactory. We only need it to point to PyPi during the Github CI/CD run, which is already taken care of with that extra step we added to the workflow. I just ran `uv sync` on this branch and it ended up just re-solving the dependencies for Artifactory because it can't access PyPi. If you're happy, I'll just commit and push that uv,lock to this branch?

Comment (coatet) [reply uv.lock:1]: I mean, I'm assuming this. When this gets published to PyPi, PyPi figures out its requirements from what's in `pyproject.toml` right?

File: CHANGELOG.md (hunk 1 above)
Comment (humnabaa) [reply CHANGELOG.md]: realised I was being too specific here. Done

File: uv.lock
Comment (humnabaa) [reply uv.lock:1]: yep, PyPi packaging is driven by pyproject.toml meta data and not by uv.lock. I have restored uv.lock to artifactory sources and kept the CI workflows

**Review by coatet — APPROVED** [review]

**Review by RHRench — APPROVED** [review]
